x-pack/plugins/security_solution/server/lib/detection_engine/rules/update_prepacked_rules.ts

```diff
@@ -128,58 +133,130 @@ export const createPromises = (
       rule: existingRule,
     });
 
-    // Note: we do not pass down enabled as we do not want to suddenly disable
-    // or enable rules on the user when they were not expecting it if a rule updates
-    return patchRules({
-      rulesClient,
-      author,
-      buildingBlockType,
-      description,
-      eventCategoryOverride,
-      falsePositives,
-      from,
-      query,
-      language,
-      license,
-      outputIndex,
-      rule: migratedRule,
-      savedId,
-      meta,
-      filters,
-      index,
-      interval,
-      maxSignals,
-      riskScore,
-      riskScoreMapping,
-      ruleNameOverride,
-      name,
-      severity,
-      severityMapping,
-      tags,
-      timestampOverride,
-      to,
-      type,
-      threat,
-      threshold,
-      threatFilters,
-      threatIndex,
-      threatIndicatorPath,
-      threatQuery,
-      threatMapping,
-      threatLanguage,
-      concurrentSearches,
-      itemsPerSearch,
-      references,
-      version,
-      note,
-      anomalyThreshold,
-      enabled: undefined,
-      timelineId,
-      timelineTitle,
-      machineLearningJobId,
-      exceptionsList,
-      throttle,
-      actions: undefined,
-    });
+    if (!migratedRule) {
+      throw new PrepackagedRulesError(`Failed to find rule ${ruleId}`, 500);
+    }
+
+    // If we're trying to change the type of a prepackaged rule, we need to delete the old one
+    // and replace it with the new rule, keeping the enabled setting, actions, throttle, id,
+    // and exception lists from the old rule
+    if (type !== migratedRule.params.type) {
+      await deleteRules({
+        ruleId: migratedRule.id,
+        rulesClient,
+        ruleExecutionLog,
+      });
+
+      return (await createRules({
```

Comment:
If any of these error out and `throw` we're just in the regular `update_prepackaged_rules` flow so it'll be an isolated failure and we'll still return an `errors[]` with any specific rule(s) that failed, right?

Reply:
If any of the updates fail, either in this new path or the existing path with `patchRules`, then it will throw all the way up to the route handler since we `await Promise.all(rulePromises);` above - that will throw if any of the promises reject. The error flow isn't changed in this PR though.